src/dmd/escape.d

```diff
@@ -804,26 +802,6 @@ ByRef:
             continue;
         }
 
-        if (va && v.isReference())
```

Comment:
Looks like this check (introduced by https://github.com/dlang/dmd/pull/7129) is redundant now that `enclosesLifetimeOf` is improved (https://github.com/dlang/dmd/pull/13530).